source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -5,4 +5,77 @@ Access Data From a Cursor
 .. default-domain:: mongodb
 
 Read operations that return multiple documents do not immediately return
-all values matching the query.
+all values matching the query. Because a query can potentially match
+large sets of documents, these operations rely upon an object called a
+:java-docs:`MongoCursor </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html>`.
+
+A ``MongoCursor`` fetches documents in batches to reduce both memory
+consumption and network bandwidth usage. 
+
+The following functions use cursors internally, and return the results of the operations:
+
+- ``collection.find()``
+- ``collection.aggregate()``
+- ``collection.listIndexes()``
+- ``db.listCollectionNames()``
+
+Append ``.cursor()`` to use these functions as cursors.  
+
+The following shows ``collection.find()`` as a cursor:
+
+.. code-block:: java
+   :copyable: true
+   
+   collection.find().cursor();
+
+Cursor Paradigms
+----------------
+
+Work with cursors using **cursor paradigms**. Cursor paradigms
+allow you to access query results one document at a time, abstracting
+away network and caching logic. 
+
+Functional Iteration
+~~~~~~~~~~~~~~~~~~~~~
+
+Pass a function to the 
+`forEachRemaining() <https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Iterator.html?is-external=true#forEachRemaining(java.util.function.Consumer)>`_ 
+method of any cursor to iterate through results in a functional style:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Cursor.java
+   :language: java
+   :dedent:
+   :start-after: begin forEachRemainingIteration
+   :end-before: end forEachRemainingIteration
+
+Manual Iteration
+~~~~~~~~~~~~~~~~
+
+Use the :java-docs:`hasNext() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html#hasNext()>`
+method to check if there are any documents available in the cursor, and then use the
+:java-docs:`next() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html#next()>`
+method to retrieve the next available document from the cursor:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Cursor.java
+   :language: java
+   :dedent:
+   :start-after: begin manualIteration
+   :end-before: end manualIteration
+
+Cursor Cleanup
+--------------
+
+Close
```

Comment:
We should show examples of calling close according to Java idioms:

1. Using try-with-resources
2. Using a finally block